src/lib/DataLoader/SearchContentLoader.php

```diff
@@ -54,16 +54,14 @@ static function (SearchHit $searchHit) {
      *
      * @param \Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion $filter A Query Criterion. Use Criterion\ContentId, Criterion\RemoteId or Criterion\LocationId for basic loading.
      *
-     * @throws \Ibexa\GraphQL\DataLoader\Exception\ArgumentsException
+     * @return \Ibexa\Contracts\Core\Repository\Values\Content\Content
+     *
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
      */
     public function findSingle(Criterion $filter): Content
     {
-        try {
```

Comment:
This seems like a behaviour change tho.

Reply:
It is, but what should be done instead? We cannot neither return an "empty content" nor make it nullable. I am open for ideas.

Reply:
While InvalidArgumentException was unhandled and resulted in a PHP error instead, the other one seems to be "handled" (by converting it into a different exception).

How does the app behave when it happens? Is it an expected user behavior to pass invalid filters?

Reply:
The only issue here is lack of return statement, which oddly appears in other classes, but was reported only in this case. 🤷 I didn't mean to change nor validate any behavior here. 

An alternative is to return null (despite it's wrong according to the interface) and put it into baseline. For obvious reasons I didn't want to do that. Changing the exception is less-aggressive though, but again - I don't have any other ideas.

Reply:
Resolved via internal sync https://github.com/ibexa/graphql/pull/68/commits/3d2ef622d6a3490b1a2bd6fa8572a817f8c6b1bd. It seems, dead exception `@throws` statements were causing the issue.

Reply:
@ViniTou after internal sync we figured out that PHPStan actually cares about `@throws` annotations, and `LocationService` have those declared. Therefore PHPStan infers that `ApiException\InvalidArgumentException` is never thrown, so it discards this execution path as never occuring.

@alongosz that might be of interest to you, as a curious behavior of PHPStan.